packages/core/notifications/core-notifications-browser-internal/src/toasts/global_toast_list.tsx

```diff
@@ -31,50 +34,76 @@ const convertToEui = (toast: ToastWithRichTitle): EuiToast => ({
   text: toast.text instanceof Function ? <MountWrapper mount={toast.text} /> : toast.text,
 });
 
-export class GlobalToastList extends React.Component<Props, State> {
-  public state: State = {
-    toasts: [],
-    idToToasts: {},
-  };
+export const GlobalToastList: FunctionComponent<Props> = ({
+  toasts$,
+  dismissToast,
+  reportEvent,
+}) => {
+  const [toasts, setToasts] = useState<State['toasts']>([]);
+  const [idToToasts, setIdToToasts] = useState<State['idToToasts']>({});
 
-  private subscription?: Subscription;
+  const reportToastDismissal = useCallback(
+    (representedToasts: State['idToToasts'][number]) => {
+      // Select the first duplicate toast within the represented toast group
+      // given it's identical to all other recurring ones within it's group
+      const {
+        0: { title: toastTitle, color: toastType },
+        length: toastRecurrenceCount,
+      } = representedToasts;
```

Comment:
Interesting that you chose to destructure this array as object instead. Do you know if there are any performance impacts of that or other gotchas?

Reply:
I benchmarked this particular case and it actually would be faster if I wasn't further destructuring the value accessed at the first index, as opposed to accessing the properties on demand. That being said I'll be switching the implementation